core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlPlanParser.scala

```diff
@@ -1,19 +1,51 @@
 package com.databricks.labs.remorph.parsers.tsql
 
-import com.databricks.labs.remorph.parsers.PlanParser
+import com.databricks.labs.remorph.intermediate.{LogicalPlan, ParsingErrors, PlanGenerationFailure}
+import com.databricks.labs.remorph.parsers.{PlanParser, ProductionErrorCollector}
 import com.databricks.labs.remorph.parsers.tsql.rules.{PullLimitUpwards, TSqlCallMapper, TopPercentToLimitSubquery, TrapInsertDefaultsAction}
-import com.databricks.labs.remorph.{intermediate => ir}
+import com.databricks.labs.remorph.{BuildingAst, KoResult, Parsing, Transformation, WorkflowStage, intermediate => ir}
 import org.antlr.v4.runtime._
+import org.antlr.v4.runtime.tree.ParseTree
 
-class TSqlPlanParser extends PlanParser[TSqlParser] {
+import scala.util.control.NonFatal
+
+class TSqlPlanParser extends PlanParser {
 
   val vc = new TSqlVisitorCoordinator(TSqlParser.VOCABULARY, TSqlParser.ruleNames)
 
-  override protected def createLexer(input: CharStream): Lexer = new TSqlLexer(input)
-  override protected def createParser(stream: TokenStream): TSqlParser = new TSqlParser(stream)
-  override protected def createTree(parser: TSqlParser): ParserRuleContext = parser.tSqlFile()
-  override protected def createPlan(tree: ParserRuleContext): ir.LogicalPlan = vc.astBuilder.visit(tree)
-  override protected def addErrorStrategy(parser: TSqlParser): Unit = parser.setErrorHandler(new TSqlErrorStrategy)
+  override def parseLogicalPlan(parsing: Parsing): Transformation[LogicalPlan] = {
```

Comment:
we need both the plan and the token stream to attach comments so had to move away from the original design